src/components/ArrowToggle/ArrowToggle.tsx

```diff
@@ -14,9 +14,56 @@ export interface ArrowToggleProps {
 
 const b = block('arrow-toggle');
 
-export function ArrowToggle({size = 16, direction = 'bottom', className}: ArrowToggleProps) {
+const transitionPeriod = 100;
+
+export function ArrowToggle({
+    size = 16,
+    direction: targetDirection = 'bottom',
+    className,
+}: ArrowToggleProps) {
+    const [direction, setDirection] = useState<ArrowToggleProps['direction'] | 'full-rotation'>(
+        targetDirection,
+    );
+    const [transitions, setTransitions] = useState(true);
+
+    useEffect(() => {
+        if (direction === targetDirection) {
+            return;
+        }
+
+        if (direction === 'full-rotation') {
```

Comment:
It's better to join this and previous condition with single return

Reply:
sure!